pkg/ebpf/tracer_windows.go

```diff
@@ -43,29 +37,52 @@ func init() {
 	}
 }
 
+// readBuffer is the buffer to pass into ReadFile system call to pull out packets
+type readBuffer struct {
+	ol   windows.Overlapped
+	data [128]byte
+}
+
 // Tracer struct for tracking network state and connections
 type Tracer struct {
-	config       *Config
-	driverHandle windows.Handle
+	config           *Config
+	driverController *DriverInterface
+	bufs             []readBuffer
+	packetCount      int64
 }
 
 // NewTracer returns an initialized tracer struct
 func NewTracer(config *Config) (*Tracer, error) {
-	handle, err := openDriverFile(driverFile)
+	dc, err := NewDriverInterface()
 	if err != nil {
-		return nil, fmt.Errorf("%s : %s", "Could not create driver handle", err)
+		return nil, fmt.Errorf("could not create windows driver controller: %v", err)
 	}
 
 	tr := &Tracer{
-		driverHandle: handle,
+		driverController: dc,
+		bufs:             make([]readBuffer, totalReadBuffers),
 	}
 
+	// We want tracer to own the buffers, but the DriverInterface to make the calls to set them up
+	tr.bufs, err = dc.prepareReadBuffers(tr.bufs)
+	if err != nil {
+		return nil, fmt.Errorf("error preparing driver's read buffers: %v", err)
+	}
+
+	err = tr.initPacketPolling()
+	if err != nil {
+		log.Warnf("issue polling packets from driver: %v", err)
+	}
```

Comment:
Is there any reason this shouldn't cause NewTracer to fail ?

Reply:
That's true - I think if there is some issue getting this set up we probably shouldn't be running the tracer. 

I'll update.